### DISCUSSION TITLE
Test Adapter or Github Integration

### DISCUSSION THREAD

**Discussion by GatlenCulp** [SanjulaGanepola/github-local-actions#130] [Ideas]
Hey! Love the tool. Two quick ideas:

1. Perhaps make a VSCode test adapter for github actions w/ act. See:

https://marketplace.visualstudio.com/items?itemName=ms-vscode.test-adapter-converter
https://code.visualstudio.com/api/extension-guides/testing

2. Perhaps duplicate the VSCode github actions extension and have it use act on the backend instead of Github. This would save a lot of work + make it consistent with Github for anyone who now wants to work locally.

https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions

**Comment by SanjulaGanepola** [comment]
Hello @GatlenCulp. Very interesting proposals.

1. While the `Testing` view could serve to replace the current `Workflows` view for executing workflows, it would not be able to render what is show in the other views. In particular the `History` view. Additionally, having 4 unique views in the extension gave me the most flexibility to implement what UI that I felt aligned with the official [GitHub Actions](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions) extension.
2. Although the UI for this extension was designed based on the official extension, there are still some differences. This is why I also wrote the code for the views from scratch. Is there anything in particular you find is in the official extension which is missing in the current extension?
